controllers/front/process.php

```diff
@@ -87,7 +129,20 @@ public function initContent()
                         'exp_month' => (int)Tools::getValue('expiration_month'),
                         'cvc' => strip_tags(trim(Tools::getValue('cvc'))),
                         'holder' => strip_tags(trim(Tools::getValue('card_holder')))
-                    )
+                    ),
+                    'threeds2_data' =>  [
+                        "notification_url" => $this->context->link->getModuleLink('cardinity', 'process'), 
+                        "browser_info" => [
+                            "accept_header" => "text/html",
+                            "browser_language" => strip_tags(trim(Tools::getValue('browser_language'))),
+                            "screen_width" => strip_tags(trim(Tools::getValue('screen_width'))),
+                            "screen_height" => strip_tags(trim(Tools::getValue('screen_height'))),
+                            'challenge_window_size' => "390x400",
```

Comment:
`challenge_window_size` parameter should be calculated dynamically.

Reply:
done